src/SDKs/SqlManagement/Sql.Tests/UploadTdeCertificateScenarioTests.cs

```diff
@@ -0,0 +1,144 @@
+        private const string ManagedInstanceResourceGroup = "MlAndzic_RG";
+        private const string ManagedInstanceName = "midemoinstance";
+        private const string InvalidPassword = "asdsadsad";
+        private const string InvalidPrivateBlobOrPasswordMessage = "Invalid private blob or password";
+
+        [Fact]
+        public void TestCreateTdeCertificate()
+        {
+            using (SqlManagementTestContext context = new SqlManagementTestContext(this))
+            {
+                ResourceGroup resourceGroup = context.CreateResourceGroup();
+                Server server = context.CreateServer(resourceGroup);
+                SqlManagementClient sqlClient = context.GetClient<SqlManagementClient>();
+
+                sqlClient.TdeCertificates.Create(resourceGroup.Name, server.Name, TdeCertificate);
+            }
+        }
+
+        [Fact]
+        public void TestCreateTdeCertificate_InvalidBlob()
+        {
+            using (SqlManagementTestContext context = new SqlManagementTestContext(this))
+            {
+                ResourceGroup resourceGroup = context.CreateResourceGroup();
+                Server server = context.CreateServer(resourceGroup);
+                SqlManagementClient sqlClient = context.GetClient<SqlManagementClient>();
+
+                var exception = Assert.Throws<CloudException>(() => sqlClient.TdeCertificates.Create(resourceGroup.Name, server.Name, TdeCertificateWithInvalidBlob));
+
+                Assert.Contains(InvalidPrivateBlobOrPasswordMessage, exception.Message);
+            }
+        }
+
+
+        [Fact]
+        public void TestCreateTdeCertificate_InvalidPassword()
+        {
+            using (SqlManagementTestContext context = new SqlManagementTestContext(this))
+            {
+                ResourceGroup resourceGroup = context.CreateResourceGroup();
+                Server server = context.CreateServer(resourceGroup);
+                SqlManagementClient sqlClient = context.GetClient<SqlManagementClient>();
+
+                var exception = Assert.Throws<CloudException>(() => sqlClient.TdeCertificates.Create(resourceGroup.Name, server.Name, TdeCertificateWithInvalidPassword));
+
+                Assert.Contains(InvalidPrivateBlobOrPasswordMessage, exception.Message);
+            }
+        }
+
+        [Fact]
+        public void TestCreateManagedInstanceTdeCertificate()
+        {
+            using (SqlManagementTestContext context = new SqlManagementTestContext(this))
+            {
+                SqlManagementClient sqlClient = context.GetClient<SqlManagementClient>();
+
+                // Update with values from a current MI on the region
+                //
+                string resourceGroup = ManagedInstanceResourceGroup;
+                string managedInstanceName = ManagedInstanceName;
```

Comment:
Tests MUST be fully automated. This means that they must create all prerequisite resources during test setup. If this is not possible, the test must be skipped (see geo restore scenario test for example)